kie-based-services/src/main/java/io/elimu/genericapi/task/StoreQRTaskEventListener.java

```diff
@@ -0,0 +1,112 @@
+package io.elimu.genericapi.task;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.jbpm.services.task.commands.TaskContext;
+import org.jbpm.services.task.events.DefaultTaskEventListener;
+import org.kie.api.task.TaskEvent;
+import org.kie.api.task.model.Task;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.Files;
+
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateExceptionHandler;
+import io.elimu.a2d2.oauth.BodyBuilder;
+import io.elimu.a2d2.oauth.OAuthUtils;
+
+public class StoreQRTaskEventListener extends DefaultTaskEventListener {
+
+	private static final Logger LOG = LoggerFactory.getLogger(StoreQRTaskEventListener.class);
+	
+	private final String fhirServerInputName;
+	private final String fhirAuthInputName;
+	private final boolean useDefaultAuth;
+
+	public StoreQRTaskEventListener(String fhirServerInputName, String fhirAuthInputName, boolean useDefaultAuth) {
+		this.fhirServerInputName = fhirServerInputName;
+		this.fhirAuthInputName = fhirAuthInputName;
+		this.useDefaultAuth = useDefaultAuth;
+	}
+
+	@Override
+	public void afterTaskCompletedEvent(TaskEvent event) {
```

Comment:
saving a questionnaire should only be performed based on the taskCode being fill-questionnaire. This event handler should have a switch based on the taskCode to call different functions (e.g., save questionnaire).